packages/yoroi-extension/app/components/wallet/send/WalletSendFormRevamp.js

```diff
@@ -513,21 +513,38 @@ export default class WalletSendForm extends Component<Props, State> {
   render(): Node {
     const { currentStep } = this.state
     return (
-      <div className={styles.component}>
-        <div className={styles.wrapper}>
-          <SendFormHeader
-            step={currentStep}
-            onUpdateStep={this.onUpdateStep.bind(this)}
-          />
-
-          <div className={styles.formBody}>
-            {this.renderCurrentStep(currentStep)}
+      <>
+        {this.renderDialog()}
+        <div className={styles.component}>
+          <div className={styles.wrapper}>
+            <SendFormHeader
+              step={currentStep}
+              onUpdateStep={this.onUpdateStep.bind(this)}
+            />
+
+            <div className={styles.formBody}>
+              {this.renderCurrentStep(currentStep)}
+            </div>
           </div>
         </div>
-      </div>
+      </>
     );
   }
 
+  renderDialog(): Node {
```

Comment:
For consistency  I think it's better to move rendering of the `QRScannerDialog` into `WalletSendPage.renderDialog`, the same way as `AddNFTDialog` and `AddTokenDialog` are rendered.

Also, as an application of the law of demeter of software design, it's better not to pass generic methods like `openDialog`, `isOpen` and `closeDialog` to this component. Instead, pass callback like `openAddNFTDialog` and `openQRScannerDialog`.

Reply:
The reason why I added the `QRScannerDialog` inside `WalletSendPage` is that I need access to the `receiverField` so I can update its state and display the address

Reply:
ok